lib/src/bootloader.rs

```diff
@@ -26,3 +26,113 @@ pub(crate) fn install_via_bootupd(
         .verbose()
         .run()
 }
+
+#[context("Installing bootloader using zipl")]
+pub(crate) fn install_via_zipl(
+    device: &Utf8Path,
+    boot_uuid: &str,
+) -> Result<()> {
+
+    // Run findmnt to find the boot partition
+
+    #[derive(serde::Deserialize)]
+    struct Filesystem {
+        source: String,
+        target: String,
+    }
+
+    #[derive(serde::Deserialize)]
+    struct Findmnt {
+        filesystems: Vec<Filesystem>,
+    }
+
+    let findmnt_desc = format!("Running findmnt to identify the boot partition with UUID={boot_uuid}");
+    let findmnt_task = Task::new(findmnt_desc, "findmnt");
+    let findmnt_args = ["--json", "--output", "TARGET,SOURCE,FSTYPE",
+                        "--uniq", "--first-only", "--nofsroot",
+                        "--source", &(format!("UUID={boot_uuid}"))];
+    let findmnt_out  = findmnt_task.args(findmnt_args).quiet().read()?;
+    let findmnt_json = serde_json::from_str::<Findmnt>(&findmnt_out).context("parsing findmnt output")?;
+
+    let boot_fs = findmnt_json.filesystems.get(0).context("finding boot filesystem")?;
+    let boot_dev = Utf8Path::new(&boot_fs.source);
+    let boot_dir = Utf8Path::new(&boot_fs.target);
+
+    // Run sfdisk to determine the boot partition offset
```

Comment:
I think we can get this data from our existing `lsblk` wrapper in `blockdev.rs` probably.

Reply:
It looks like we can use the function in blockdev.rs. I'll update the code.

Reply:
We can probably change it to run `lsblk -O -J` to basically scrape all the data by default instead of the explicit current allowlist.